internal/cache/contentcache.go

```diff
@@ -0,0 +1,162 @@
+package cache
+
+import (
+	"fmt"
+	"sync"
+	"time"
+	"net/http"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+type ContentCache struct {
+	m     *mm
+	size  int64
+	cache *ristretto.Cache
+	debug bool
+}
+
+type Response struct {
+	Header     http.Header
+	Body       []byte
+	StatusCode int
+}
+
+func NewContentCache(contentCacheSize int64, debug bool) (*ContentCache, error) {
+	m := New()
+	size := contentCacheSize
+	nc := size / 1000
+	if nc < 100 { //mainly for testing
+		nc = 100
+	}
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		//NumCounters is 10 times estimated max number of items in cache, as suggested in https://pkg.go.dev/github.com/dgraph-io/ristretto@v0.1.1#Config
+		NumCounters:        nc, //limit / 10 KB small files * 10
+		MaxCost:            size,
+		BufferItems:        64,
+		Metrics:            debug,
+		IgnoreInternalCost: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &ContentCache{m: m, size: size, cache: cache, debug: debug}, nil
+}
+
+func (c *ContentCache) Get(key string) (*Response, bool) {
+	l := c.m.RLock(key)
+	defer l.RUnlock()
```

Comment:
`ristretto.Cache` operations (`Get` & `Set`) are thread-safe, so we don't need extra locking.